### PR TITLE
dropFile

### PR DESCRIPTION
**Pull request by faust64**
dropFile method

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -123,6 +123,33 @@ S3Client.prototype.storeFile = function (options, callback) {
...
+        headers  = options.headers || {},
+        s3 = this.s3;
+
+    return new Promise(function dropFilePromise(resolve, reject) {
+        s3.del(filename, headers)
+            .on('response', function dropFileResponse(response) {
+                if (response.statusCode !== 200) {
+                    return reject(new Error('Failed to delete'));
+                }
+                return;
```
Comment (AVVS) [inline lib/client.js]: must call `response.resume()` or `response.pipe` or `response.on('data')` to comsume body and not stall socket open

File: lib/client.js [hunk 2]
```diff
@@ -123,6 +123,33 @@ S3Client.prototype.storeFile = function (options, callback) {
...
+            })
+            .on('error', reject)
+            .end(function dropFileReturn() {
```
Comment (AVVS) [inline lib/client.js]: this is what's send to the upstream, therefore it doesn't belong here

File: lib/client.js (hunk 1 above)
Comment (AVVS) [inline lib/client.js]: resolve after one of the actions outlined

**Comment by faust64** [comment]
I'll need more tests. Will get back to you.

**Pull request closed without merging**